### ISSUE TITLE
The last (first) result is always nil

### ISSUE THREAD

**Issue by walseb** [squiter/ivy-youtube#15]
The last (as currently sorted by `ivy-youtube` #14 ) search result is always nil. This is probably caused by `ivy-youtube-wrapper` or `ivy-youtube-tree-assoc` since `(ivy-youtube-tree-assoc 'items *qqJson*)` does contain the entry that's nil. This problem is present in all `ivy-youtube` versions i have tried (initial commit, before my recent changes, master)

**Comment by squiter** [comment]
Maybe we can fix it with a simple `(remove nil '(nil 1 nil 2 nil 3 nil 4))`... :thinking:    
I'll try to fix it this week

**Comment by walseb** [comment]
Yeah but then it will not display the most relevant video, which is bad if you search for a particular video since it will never show up.
Anyway, the problem must be with `(cl-loop for x being the elements of (ivy-youtube-tree-assoc 'items *qqJson*)` because doing
```
    (cl-loop for x being the elements of (ivy-youtube-tree-assoc 'items *qqJson*)
             do
	     (add-to-list 'test-variable x) ;; <- The only change I made to debug
	     (push (cons (cdr (ivy-youtube-tree-assoc 'title x))
                         (cdr (ivy-youtube-tree-assoc 'videoId x))) *results*))
```
results in `test-variable` being a list with only 49 title entries and every title is there except the one that's normally nil. Also ` (ivy-youtube-tree-assoc 'items *qqJson*)` has 50 correctly formatted title entries so the problem can't be there. 
I'm not sure exactly how cl-loops works, maybe you know how to fix this loop?

**Comment by squiter** [comment]
Fixed in #16
